main.go

```diff
@@ -365,7 +365,10 @@ func (s *DB) Rollback() *DB {
 }
 
 func (s *DB) NewRecord(value interface{}) bool {
-	return s.clone().NewScope(value).PrimaryKeyZero()
+	scope := s.clone().NewScope(value)
+	result := scope.PrimaryKeyZero()
+	s.err(scope.db.Error)
```

Comment:
Without it I don't think the unit-test will pass.  I will double check and provide more details when I get home.



> On Aug 12, 2015, at 7:09 AM, Jinzhu <notifications@github.com> wrote:
> 
> In main.go:
> 
> > @@ -365,7 +365,10 @@ func (s *DB) Rollback() *DB {
> >  }
> >  
> >  func (s *DB) NewRecord(value interface{}) bool {
> > -	return s.clone().NewScope(value).PrimaryKeyZero()
> > +	scope := s.clone().NewScope(value)
> > +	result := scope.PrimaryKeyZero()
> > +	s.err(scope.db.Error)
> This s.err() is not necessary, there will be SQL be invoked here.
> 
> —
> Reply to this email directly or view it on GitHub.
>